pkg/proxy/proxy.go

```diff
@@ -289,9 +289,8 @@ func (p *Proxy) ackProxyPort(name string, pp *ProxyPort) error {
 		// Add rules for the new port
 		// This should always succeed if we have managed to start-up properly
 		scopedLog.Infof("Adding new proxy port rules for %s:%d", name, pp.proxyPort)
-		err := p.datapathUpdater.InstallProxyRules(pp.proxyPort, pp.ingress, name)
-		if err != nil {
-			return fmt.Errorf("Cannot install proxy rules for %s: %s", name, err)
+		if err := p.datapathUpdater.InstallProxyRules(pp.proxyPort, pp.ingress, name); err != nil {
```

Comment:
I think `Install{,Proxy}Rules` should take the daemon `context` so that we can stop the retries if they're in the middle of executing when the daemon is shutting down. 

See calls to `d.Datapath().Loader().Reinitialize()` for reference.

Reply:
added a `context` to both `InstallRules` and `InstallProxyRules` :+1: 
The latter was a bit involved as it's called also from `Proxy.ackProxyPort()`, which in turn is called from a few different places